cmd/gitops/beta/run/cmd.go

```diff
@@ -50,6 +74,266 @@ func betaRunCommandPreRunE(endpoint *string) func(*cobra.Command, []string) erro
+
 		return nil
 	}
 }
+
+func getFluxVersion(obj unstructured.Unstructured) (string, error) {
+	labels := obj.GetLabels()
+	if labels == nil {
+		return "", fmt.Errorf("error getting labels")
+	}
+
+	fluxVersion := labels[flux.VersionLabelKey]
+	if fluxVersion == "" {
+		return "", fmt.Errorf("error getting flux version")
+	}
+
+	return fluxVersion, nil
+}
+
+func installFlux(kubeClient ctrlclient.Client, ctx context.Context, filePath string, shortFilename string) error {
+	fmt.Println("filepath:", filePath)
+	fmt.Println("shortFilename:", shortFilename)
+
+	opts := install.Options{
+		BaseURL:      install.MakeDefaultOptions().BaseURL,
+		Version:      "v0.31.2",
+		Namespace:    "flux-system",
+		Components:   []string{"source-controller", "kustomize-controller", "helm-controller", "notification-controller"},
+		ManifestFile: "flux-system.yaml",
+		Timeout:      5 * time.Second,
+	}
+
+	manifest, err := install.Generate(opts, "")
+	if err != nil {
+		return fmt.Errorf("couldn't generate manifests: %+v", err)
+	}
+
+	content := []byte(manifest.Content)
+
+	err = os.MkdirAll(filePath, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("couldn't create file %+v", err)
+	}
+
+	err = os.WriteFile(filepath.Join(filePath, shortFilename), content, 0666)
+	if err != nil {
+		return fmt.Errorf("couldn't write flux manifests to file %+v", err)
+	}
+
+	rf := rootFlags{
+		pollInterval: 2 * time.Second,
+		defaults:     install.MakeDefaultOptions(),
+	}
+
+	var VERSION = "0.0.0-dev.0"
+
+	rf.defaults.Version = "v" + VERSION
+
+	var kubeconfigArgs = genericclioptions.NewConfigFlags(false)
+
+	var kubeclientOptions = new(runclient.Options)
+
+	// *kubeconfigArgs.Namespace = rootArgs.defaults.Namespace
+	fromEnv := os.Getenv("FLUX_SYSTEM_NAMESPACE")
+	if fromEnv != "" {
+		kubeconfigArgs.Namespace = &fromEnv
+	}
+
+	kubeconfigArgs.APIServer = nil // prevent AddFlags from configuring --server flag
+	kubeconfigArgs.Timeout = nil   // prevent AddFlags from configuring --request-timeout flag, we have --timeout instead
+	// kubeconfigArgs.AddFlags(rootCmd.PersistentFlags())
+
+	// Since some subcommands use the `-s` flag as a short version for `--silent`, we manually configure the server flag
+	// without the `-s` short version. While we're no longer on par with kubectl's flags, we maintain backwards compatibility
+	// on the CLI interface.
+	apiServer := ""
+	kubeconfigArgs.APIServer = &apiServer
+	// rootCmd.PersistentFlags().StringVar(kubeconfigArgs.APIServer, "server", *kubeconfigArgs.APIServer, "The address and port of the Kubernetes API server")
+
+	// kubeclientOptions.BindFlags(rootCmd.PersistentFlags())
+
+	// rootCmd.RegisterFlagCompletionFunc("context", contextsCompletionFunc)
+	// rootCmd.RegisterFlagCompletionFunc("namespace", resourceNamesCompletionFunc(corev1.SchemeGroupVersion.WithKind("Namespace")))
```

Comment:
What should I do with global flags (if any of those supported by the `gitops` command are provided) in the `run` command here? I've just copy-pasted and commented out code from the Flux instal command.

Which of this or original `gitops` flags do I need to add here?

Reply:
Oh god, do I have to make decisions :scream: 

So, I looked at what flags `flux install --help` and `gitops --help` prints out and tried to look at what's important Note: this is written without looking at the code, so if I say something that is not fairly straight-forward copy-paste, then the correct answer is "that's very interesting Robin, I suggest you make a ticket if you feel strongly about it, because that's not part of this ticket's description".

`--version` makes sense - especially as that'd let us set and print a default in the help text, so it's possible to figure out what we'd do. We should have a default set at compiletime in `pkg/version/version.go` already. Maybe it should be called `--flux-version` instead here, though. 

I think maybe `--components-extra` and `--components` would be useful as well, again as a bit of documentation in the help text what the thing will do.

There's already a global `--namespace` in gitops, so we should use that here - especially since I've come to grow opinions about namespaces and gitops recently :blush: 

We need a timeout set to _something_ or install doesn't work, and what I hard-coded isn't a good timeout. Let's just add a `--timeout` flag?

`--insecure-skip-tls-verify` is a global arg in both gitops and flux, is it easy to hook up?

The rest of the global gitops flags are irrelevant.

There's _loads_ of global flux flags which I'm sure are important to some people (personally, I really want `--context`), but they all require changes to the kube config/client bits, so I think they're out of scope for now.